ocaml/xapi/xapi_http.ml

```diff
@@ -199,7 +213,11 @@ let with_context ?(dummy = false) label (req : Request.t) (s : Unix.file_descr)
         | None, None, Some (Http.UnknownAuth x) ->
             raise (Failure (Printf.sprintf "Unknown authorization header: %s" x))
         | None, None, None ->
-            raise (Http.Unauthorised label)
+            let session_id =
+              try create_session_for_client_cert req s
+              with _ -> raise (Http.Unauthorised label)
+            in
+            (session_id, false)
```

Comment:
Here, `must_logout` should be `true`, as no one else will logout (the client does not get this session id).

Reply:
Yes. Thanks. Fixed it.